packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,93 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
```

Comment:
When I navigate to the root url without a locale such as https://magento-venia-concept-mhkx8.local.pwadev:8389/ I get a warning:

[![Image from Gyazo](https://i.gyazo.com/eda94887070c51137bda5f0cc26596cc.png)](https://gyazo.com/eda94887070c51137bda5f0cc26596cc)

Reply:
This is expected, the warning is a bit annoying but it mimics the Luma functionality here. You can still visit pages without the store code, but every link will get the store code included when you navigate. 

I believe this warning won't occur on a production system, I don't know if we should try suppress this during development?

Reply:
Shouldn't warn in production. Worth double checking @dpatil-magento